pkg/occlient/occlient_test.go

```diff
@@ -406,3 +408,883 @@ func Test_parseImageName(t *testing.T) {
+					Template: &corev1.PodTemplateSpec{
+						ObjectMeta: metav1.ObjectMeta{
+							Labels: map[string]string{
+								"deploymentconfig": "wildfly",
+							},
+						},
+						Spec: corev1.PodSpec{
+							Containers: []corev1.Container{
+								{
+									Image: "wildfly:latest",
+									Name:  "wildfly",
+								},
+							},
+						},
+					},
+				},
+			},
+			createdPVC: corev1.PersistentVolumeClaim{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: fmt.Sprintf("%s-s2idata", "wildfly"),
+					Labels: map[string]string{
+						"app": "apptmp",
+						"app.kubernetes.io/component-name": "wildfly",
+						"app.kubernetes.io/component-type": "wildfly",
+						"app.kubernetes.io/name":           "apptmp",
+					},
+				},
+				Spec: corev1.PersistentVolumeClaimSpec{
+					Resources: corev1.ResourceRequirements{
+						Requests: corev1.ResourceList{
+							corev1.ResourceStorage: quantity,
+						},
+					},
+					AccessModes: []corev1.PersistentVolumeAccessMode{
+						corev1.ReadWriteOnce,
+					},
+				},
+			},
+
+			wantErr: false,
+		},
+		{
+			name:        "dc name with annotations",
+			dcName:      "wildfly",
+			projectName: "project-testing",
+			annotations: map[string]string{
+				"app.kubernetes.io/url":                   "file:///temp/nodejs-ex",
+				"app.kubernetes.io/component-source-type": "local",
+			},
+			labels: map[string]string{
+				"app": "apptmp",
+				"app.kubernetes.io/component-name": "ruby",
+				"app.kubernetes.io/component-type": "ruby",
+				"app.kubernetes.io/name":           "apptmp",
+			},
+			existingDc: appsv1.DeploymentConfig{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "wildfly",
+					Annotations: map[string]string{
+						"app.kubernetes.io/url":                   "https://github.com/sclorg/nodejs-ex",
+						"app.kubernetes.io/component-source-type": "git",
+					},
+				},
+				Spec: appsv1.DeploymentConfigSpec{
+					Template: &corev1.PodTemplateSpec{
+						ObjectMeta: metav1.ObjectMeta{
+							Labels: map[string]string{
+								"deploymentconfig": "wildfly",
+							},
+						},
+						Spec: corev1.PodSpec{
+							Containers: []corev1.Container{
+								{
+									Image: "wildfly:latest",
+									Name:  "wildfly",
+								},
+							},
+						},
+					},
+				},
+			},
+			createdPVC: corev1.PersistentVolumeClaim{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "wildfly",
+					Labels: map[string]string{
+						"app": "apptmp",
+						"app.kubernetes.io/component-name": "wildfly",
+						"app.kubernetes.io/component-type": "wildfly",
+						"app.kubernetes.io/name":           "apptmp",
+					},
+				},
+				Spec: corev1.PersistentVolumeClaimSpec{
+					Resources: corev1.ResourceRequirements{
+						Requests: corev1.ResourceList{
+							corev1.ResourceStorage: quantity,
+						},
+					},
+					AccessModes: []corev1.PersistentVolumeAccessMode{
+						corev1.ReadWriteOnce,
+					},
+				},
+			},
+
+			wantErr: true,
+		},
+		{
+			name:        "dc name with annotations",
```

Comment:
Need a different name for the test. Also, is this a negative case because of mismatch in `dcName` and `existingDC`?  I am not sure if that's a valid case for us then